deploy_sphinx_docs_multiversion/action.yml

```
@@ -0,0 +1,164 @@
+name: 'Deploy Sphinx Docs Multi_version'
+description: 'Deploys pre-built sphinx docs with versioning on GitHub Pages'
+
+inputs: 
+  secret_input:
+    description: 'The secret input for the GitHub token'
+    required: true
+  use-make:
+    description: |
+      Specify whether `make` is used in the `build_sphinx_docs` action,
+      so that the correct publish directory is used
+    required: false
+    type: boolean
+    default: false
+  switcher-url:
+    description: |
+      The URL to the switcher.json file that will be updated with the new version.
+    required: true
+  base-url:
+    description: |
+      The URL to be used as the base-url for the initialising the switcher.json file.
+    required: false
+    default: ""
+
+runs:
+  using: 'composite'
+  steps:
+  - name: Checkout repository
+    uses: actions/checkout@v4
+
+  - name: Clear previous build if present
+    shell: bash
+    run: rm -rf ./docs/build
+
+  - name: Download the content for deployment
+    uses: actions/download-artifact@v4
+    with:
+      name: docs-build
+      path: ./docs/build
+
+  - name: Checkout the gh-pages branch in a separate folder
+    uses: actions/checkout@v4
+    with:
+      ref: gh-pages
+      # Checkout to this folder instead of the current one
+      path: deploy
+      # Download the entire history
+      fetch-depth: 0
+
+  - name: Push the built HTML to gh-pages
+    shell: bash 
+    run: |
+      # Detect if this is a release or from the main branch
+      echo "Event name: ${{ github.event_name }}"
+      echo "Ref type: ${{ github.ref_type }}"
+      if [ "${{ github.event_name }}" = "push" ] && [ "${{ github.ref_type }}" = "tag" ]; then
+          # Get the tag name without the "refs/tags/" part
+          version="${GITHUB_REF#refs/*/}"
+      else
+          version=dev
+      fi
+
+      echo "Deploying version: $version"
+      # Make the new commit message. Needs to happen before cd into deploy
+      # to get the right commit hash.
+      message="Deploy $version from $(git rev-parse --short HEAD)"
+      cd deploy
+      # Need to have this file so that Github doesn't try to run Jekyll
+      touch .nojekyll
+
+      #Add redirection from root url to latest
```

Added a space after # for consistency with other comments. Apologies for being pedantic
```suggestion
      # Add redirection from root url to latest
```